modules/kafka/kafka-dest-worker.c

```diff
@@ -335,7 +367,15 @@ _set_methods(KafkaDestWorker *self)
 
   if (owner->transaction_commit)
     {
-      self->super.insert = kafka_dest_worker_transactional_insert;
+      if (owner->super.batch_lines > 0)
```

Comment:
Maybe we could add a warning in case the user wants to configure batching while the transaction API is not used ( case of sync_send(no)).

Reply:
We warn the user in http() driver if the number of workers are less than the number of URLs.